app/models/optimization.rb

```diff
@@ -154,4 +154,34 @@ def log_error(err_msg)
   def raise_error(err_msg)
     raise OptimizationError.new(err_msg)
   end
+
+  def nb_points
+    self.inputs.empty? ? "0" : self.inputs["x"].length
+  end
+
+  def status_text
```

Comment:
Actually this kind of methods are helpers for views and are rather located in a Ruby module under `app/helpers`.